radis/lbl/base.py

```diff
@@ -999,17 +1006,24 @@ def _add_EvibErot_RADIS_cls1(self, df, calc_Evib_harmonic_anharmonic=False):
             t0 = time()
 
         # Check energy levels are here
-        for iso in self._get_isotope_list():
-            if not iso in self.parsum_calc[molecule]:
-                raise AttributeError(
-                    "No Partition function calculator defined for isotope {0}".format(
-                        iso
+        try:
+            for iso in self._get_isotope_list():
+                if not iso in self.parsum_calc[molecule]:
```

Comment:
Works, but I find it hard to read, as we're testing two things at the same time (loading of energies, and existence for a particular isotope).

What about replacing : 
```
                if not iso in self.parsum_calc[molecule]:
```
with  : 
```
if not iso in get_partition_function_calculator(self, molecule, isotope, "X")
```
(here "X" refers to the main electronic state, it should be defined this way).

If energies haven't been loaded, we'll trigger your new error message from loader.py). No need for the try/except in base.py anymore.

Reply:
Actually, there is one unusual error I'm facing while doing this. For Test - `test_calc_hitemp_CO_noneq` we are computing - 
```
calc_spectrum(
    wavenum_min=2000 ,
    wavenum_max=2300 ,
    molecule="CO",
    isotope="1,2",
    Tvib=1500,
    Trot=300,
    databank="hitemp",  # test by fetching directly
)
```
I'm getting a Type Error - 
```
molecule = self.input.molecule
state = self.input.state  # electronic state

for iso in self._get_isotope_list():
>               if not iso in self.get_partition_function_calculator(self, molecule, iso, state):
E               TypeError: get_partition_function_calculator() takes 4 positional arguments but 5 were given

radis/lbl/base.py:1011: TypeError
```
When printing the values of all parameters we get - 
```
molecule -  CO 
isotope -  1 
elec-state -  X
```
Which seems appropriate, what can be causing the additional argument?
There is a TODO for the above - 
# TODO: for multi-molecule mode: add loops on molecules and states too. This can justify the cause of the error, but the parameters are single values only in the above example, so what can be the culprit?

Reply:
My bad, I wrongly copy-pasted the function name. Should be:  
```
if not iso in self.get_partition_function_calculator(molecule, iso, state):
```
not 
```
if not iso in self.get_partition_function_calculator(**self**, molecule, iso, state):<
```

Reply:
This time I got the following error -
```
for iso in self._get_isotope_list():
                print("molecule - ", molecule, "\nisotope - ",iso, "\nelec-state - ", state)
>               if not iso in self.get_partition_function_calculator(molecule, iso, state):
TypeError: argument of type 'PartFunc_Dunham' is not iterable
radis/lbl/base.py:1012: TypeError
```
This may be due to the fact that we want -
if not iso in `self.parsum_calc[molecule]`

But `self.get_partition_function_calculator(molecule, iso, state)` returns `self.parsum_calc[molecule][isotope][elec_state]`. We actually need the superset of parsum ie. `self.parsum_calc[molecule]` which can't be derived from a subset `self.parsum_calc[molecule][isotope][elec_state]`, which could be true the other way around.

Reply:
You're right, let's keep it as you suggested initially, unless you see another way!